control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -191,6 +191,13 @@ impl ResourceResize for OperationGuardArc<VolumeSpec> {
         let spec = self.as_ref().clone();
         let state = registry.volume_state(&request.uuid).await?;
 
+        // Pre-check - Don't allow resize if the volume has snapshots.
+        // NOTE: This is temporary till the fix for handling this in spdk is done.
+        if !registry.specs().snapshots_by_vol(self.uuid()).is_empty() {
+            return Err(SvcError::Internal {
```

Comment:
@Abhinandan-Purkait will Internal "avoid" the storm of retries or we need a different error code?

Reply:
```
codes.Canceled,          // gRPC: Client Application cancelled the request
codes.DeadlineExceeded,  // gRPC: Timeout
codes.Unavailable,       // gRPC: Server shutting down, TCP connection broken - previous volume operation may be still in progress.
codes.ResourceExhausted, // gRPC: Server temporarily out of resources - previous volume operation may be still in progress.
codes.Aborted:           // CSI: Operation pending for volume
```

Only these error codes will do exponential backoff, all others including internal will be burst retries.

Reply:
`Aborted` may be used here then I guess. Not a perfect fit, but will be ok.

Reply:
Will map `Aborted` in the CSI controller, let's keep Internal here.